### PR TITLE
Fix timing problems in test suite on OS X et al

### PR DESCRIPTION
**Pull request by 2shortplanks**

### CODE REVIEW

File: t/Runner-no-unnecessary-rebuild.t [hunk 1]
```diff
@@ -169,3 +205,34 @@ EOF
...
 done_testing();
+
+########################################################################
+
+sub touch_and_ensure_new_mtime {
+    my $file = shift;
+
+    # the simple case, where the file system has sub-second modification times
+    my $mtime = $file->stat->mtime;
+    sleep 0.01;
```
Comment (oschwald) [inline t/Runner-no-unnecessary-rebuild.t]: Can't you just use `utime`?

File: t/Runner-no-unnecessary-rebuild.t [hunk 2]
```diff
@@ -92,6 +111,11 @@ my $iteration = 1;
 {
     my $runner = Stepford::Runner->new(
         step_namespaces => 'Test::Step',
+        # logger          => Log::Dispatch->new(
```
Comment (autarch) [inline t/Runner-no-unnecessary-rebuild.t]: Please remove this commented-out code.

File: t/Runner-no-unnecessary-rebuild.t [hunk 3]
```diff
@@ -169,3 +205,34 @@ EOF
...
+    my $file = shift;
+
+    # the simple case, where the file system has sub-second modification times
+    my $mtime = $file->stat->mtime;
+    sleep 0.01;
+    $file->touch;
+    return if $mtime != $file->stat->mtime;
+
+    # in *theory* we could use utime here to do things, but that comes with
+    # its own portability issues when I looked into it
```
Comment (oschwald) [inline t/Runner-no-unnecessary-rebuild.t]: Sorry, just saw this. What portability issues? Windows?

Comment (2shortplanks) [inline t/Runner-no-unnecessary-rebuild.t]: Summary of slack discussion: I'm coding defensively here, and there's several issues with utime:

- It either involves setting the modification time of files in the future or setting all the existing files to some time in the past
- There's portability issues with NFS (and presumably other esoteric systems), server time verses local time, permissions models, etc.

The sleep in this situation isn't a _blind_ sleep, but rather a sleep for `0.01` seconds repeatedly until something changes, so it's (a) as quick as can be and (b) shouldn't be effected by timing problems on a busy system.

**Comment by autarch** [comment]
I merged this from the CLI.

**Pull request closed without merging**